### PR TITLE
GY-SR04 sonar range finder driver

### PR DESCRIPTION
**Pull request by SalimTerryLi**
**Describe problem solved by this pull request**
This PR adds support for GY-SR04 sonar range finder.

**Describe your solution**
GY-US42 supports UART, IIC and level trigger modes. This driver only add the support for IIC, which is easy to adopt and has less overhead compared to UART.

**Describe possible alternatives**
Although this module is very similar to SRF02 and have almost the same address map, those two drivers won't work for other and I have no idea how to distinguish them. So this new driver comes.

**Test data / coverage**
https://logs.px4.io/plot_app?log=7f8717c4-a16a-4f13-90ee-7055bd28b170

**Additional context**
TODO

### CODE REVIEW

File: src/drivers/distance_sensor/gy_us42/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,43 @@
+############################################################################
+#
+#   Copyright (c) 2015-2020 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/drivers/distance_sensor/gy_us42/CMakeLists.txt:3]:
```suggestion
#   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```

File: src/drivers/drv_sensor.h [hunk 2]
```diff
@@ -165,6 +165,7 @@
 
 #define DRV_ADC_DEVTYPE_ADS1115	0x90
 #define DRV_DIST_DEVTYPE_VL53L1X 0x91
+#define DRV_DIST_DEVTYPE_GY_US42 0x9f
```
Comment (dagar) [inline src/drivers/drv_sensor.h:168]:
```suggestion
#define DRV_DIST_DEVTYPE_GY_US42 0x92
```

Is there a reason it jumps forward? It doesn't really matter yet until we start really using device IDs for range finders, but I'd like to try and minimize gaps.

**Comment by dagar** [comment]
Good to go?

File: src/drivers/distance_sensor/gy_us42/CMakeLists.txt (hunk 1 above)
Comment (dagar) [inline src/drivers/distance_sensor/gy_us42/CMakeLists.txt:3]:
```suggestion
#   Copyright (c) 2021 PX4 Development Team. All rights reserved.
```

**Comment by SalimTerryLi** [comment]
> Good to go?

The driver works without any problems but, I haven't done any test that actually use baro data. Expecting a flight with altitude relays on baro sensor to prove it's usability.

**Comment by SalimTerryLi** [comment]
I think it is OK to go. Although I'm not able to perform further testes but the sensor itself runs with no problem.
https://logs.px4.io/plot_app?log=f5b551e3-3f75-40ce-b07a-4f688b762323
@dagar

**Review by dagar — APPROVED** [review]

**Pull request merged**